Updates to App Badging API (#3617)
GoogleChrome/web.dev: src/site/content/en/blog/badging-api/index.md

`setAppBadge(unreadCount)` and let the user agent deal with
displaying it accordingly.

While the App Badging API *in Chrome* requires an installed app, you shouldn't
make calls to the Badging API dependent on the install state. Just call the
API when it exists, as other browsers may show the badge in other places.
If it works, it works. If not, it simply doesn't.

## Setting and clearing the badge in the background from a service worker

> In addition to setting the app badge while the page is open, it can also be
> set in the background by the service worker for the page, allowing it to be
> updated, even when the app isn't open. This can be done either through the

Reviewer: ```suggestion
You can set app badges form more than the open page. You can also set them in
the background using the service worker, allowing them to be update even when
the app isn't open. Do this either through the
```

Reviewer: The problem I'm trying to correct is the shift in actor that happens between the subordinate and main clause. The subordinate clause that introduces the sentence implies an actor other than the app badge. In the main clause, the app badge is the actor. This becomes clear if you substitute the antecedent for the pronoun 'it'.

>In addition to setting the app badge, the app badge can also be set...

I'm not in love with my wording. If you have a better idea, I would only encourage you to stick with active voice.

Author: Thanks, rewrote this section.

Push API, periodic background sync, or a combination of both.

### Periodic background sync

[Periodic background sync](/periodic-background-sync/) allows a service worker
to periodically poll the server, which could be used to get an updated status,
and call `navigator.setAppBadge()`.

However, the frequency at which the sync is called isn't perfectly reliable,
and is called the at discretion of the browser.

### Web Push API

The [Push API][push-api] allows servers to send messages to service workers,
which can run JavaScript code even when no foreground page is running. Thus,
a server push could update the badge by calling `navigator.setAppBadge()`.

However, most browsers, Chrome included, require a notification to be

> displayed whenever a push message is received. This is fine for some uses

Reviewer: ```suggestion
displayed whenever a push message is received. This is fine for some use
```

> cases, for example if you always want to show a notification when updating
> the badge. But, it makes it impossible to subtly update the badge without

Reviewer: ```suggestion
cases (for example showing a notification when updating
the badge) but makes it impossible to subtly update the badge without
```

displaying a notification.

In addition, users must grant your site notification permission in order to
receive push messages.

### A combination of both

> While not perfect, the use of Push API and periodic background sync together

Reviewer: ```suggestion
While not perfect, using Push API and periodic background sync together
```

provide a good solution. High priority information is delivered via the Push
API, showing a notification and updating the badge. And lower priority
information is delivered by updating the badge, either when the page is open,
or via periodic background sync.

### The future

The Chrome team is investigating ways to more reliably [update the app badge in
the background](https://github.com/w3c/badging/blob/master/explainer.md#background-updates),

> and would like to hear from you. Let them know what would work best for your

Reviewer: ```suggestion
and want to hear from you. Let them know what works best for your
```
